### PR TITLE
Instructor Teaching Call Form Typeahead improvements

### PR DESCRIPTION
**Pull request by ltwheeler**
Changes:
1) Hitting enter while typing will no longer select the first result and add it as a preference.
2) Typeahead is much more intelligent at parsing queries, should have less user frustration at lack of matches
3) No longer driven by DW, uses only scheduled courses. Suggested courses must now be in comment box.

### CODE REVIEW

**Comment by cthielen** [comment]
Any chance you could throw the course names in the dropdown results as part of this? It's another issue that's open that could be a real easy win if the data is already in the payload.

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js [hunk 1]
```diff
@@ -18,53 +18,74 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
...
+						distance: 100,
+						maxPatternLength: 32,
+						minMatchCharLength: 1,
+						includeScore: false,
+						keys: [
+							"description"
+						]
+					};
+
+					var fuse = new Fuse(termContainer.preferenceOptions, options);
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: Does Fuse() need to be cleaned up? Does the JS garbage collector handle this?

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js [hunk 2]
```diff
@@ -18,53 +18,74 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
...
-							return 1;
+			// Will improve query formatting when possible to improve search score
+			// For example if it receives 'ECS 10' it will format it as 'ECS 010'
+			$scope.optimizeQueryFormat = function(query) {
+				var optimizedQuery = angular.copy(query);
+
+				// Is there a space?
+				if (optimizedQuery.indexOf(' ') == -1) {
+					// Does it have 6 chars?
+					if (optimizedQuery.length == 6){
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: Missing space between ) and {

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js [hunk 3]
```diff
@@ -18,53 +18,74 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
...
-
 			};
 
-			$scope.sortCourses = function(courses) {
-					courses.sort(function (a, b) {
-						// Use subject codes to sort if they don't match
-						if (a.subjectCode > b.subjectCode) {
-							return 1;
+			// Will improve query formatting when possible to improve search score
+			// For example if it receives 'ECS 10' it will format it as 'ECS 010'
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: Does it handle other cases? If so, could you list them here?

**Comment by cthielen** [comment]
Definitely works much better, thank you!

File: app/teachingCall/teachingCall.html [hunk 4]
```diff
@@ -60,6 +60,7 @@
 	<script src="/js/snippets.js"></script>
 	<script src="/js/teachingCallApp.js"></script>
 	<script src="/js/jquery-ui.min.js"></script>
+	<script src="/js/fuse-3.0.4.min.js"></script>
```
Comment (cthielen) [inline app/teachingCall/teachingCall.html]: Is there an npm package instead? I'd prefer we use modern package management.

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js (hunk 1 above)
Comment (cthielen) [reply app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: Nvm, there's no cleanup method in their API.

**Comment by ltwheeler** [comment]
Awaiting feature review before merging.

**Pull request merged**